node/overseer/subsystems-gen/src/lib.rs

```diff
@@ -0,0 +1,188 @@
+// Copyright 2021 Parity Technologies (UK) Ltd.
+// This file is part of Polkadot.
+
+// Polkadot is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Polkadot is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Polkadot.  If not, see <http://www.gnu.org/licenses/>.
+
+use std::collections::HashSet;
+
+use proc_macro2::TokenStream;
+use quote::quote;
+
+use syn::{AttrStyle, Error, GenericParam, Ident, Result, Type, parse2};
+
+#[proc_macro_derive(AllSubsystemsGen, attributes(sticky))]
```

Comment:
Are we planning to use `sticky` attribute in `AllSubsystems`? I don't think so (YAGNI).